sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -360,13 +362,15 @@ object ViewHelper {
   def generateViewProperties(
       properties: Map[String, String],
       session: SparkSession,
-      analyzedPlan: LogicalPlan): Map[String, String] = {
+      analyzedPlan: LogicalPlan,
+      fieldNames: Array[String]): Map[String, String] = {
+    // for createViewCommand queryOutput may be different from fieldNames
     val queryOutput = analyzedPlan.schema.fieldNames
 
     // Generate the query column names, throw an AnalysisException if there exists duplicate column
     // names.
     SchemaUtils.checkColumnNameDuplication(
```

Comment:
we can make the diff much smaller by changing this line to
```
val colNamesToCheck = if (userSpecifiedColumns.nonEmpty) userSpecifiedColumns.map(_._1) else queryOutput
SchemaUtils.checkColumnNameDuplication(colNamesToCheck, ...)
```

Reply:
> we can make the diff much smaller by changing this line to
> 
> ```
> val colNamesToCheck = if (userSpecifiedColumns.nonEmpty) userSpecifiedColumns.map(_._1) else queryOutput
> SchemaUtils.checkColumnNameDuplication(colNamesToCheck, ...)
> ```
userSpecifiedColumns is not accessible in SchemaUtils. I guess we will have to pass them as arguments

Reply:
We only need to access `userSpecifiedColumns` before calling `SchemaUtils`, don't we?

Reply:
My bad, I meant `userSpecifiedColumns` is not available in `ViewHelper` not `SchemaUtils`

- `CreateViewCommand.prepareTable()` and `AlterViewAsCommand.alterPermanentView()` both call  `ViewHelper.generateViewProperties()`
- And `ViewHelper.generateViewProperties()` is where we call `SchemaUtils.checkColumnNameDuplication()`. The suggestion you provided requires `userSpecifiedColumns` to be accessible in `ViewHelper.generateViewProperties()`
- So i added an argument to `ViewHelper.generateViewProperties()` allowing `CreateViewCommand` to send aliased schema and `AlterViewAsCommand` to send analyzed schema